pyproject.toml

```
@@ -25,17 +25,50 @@ abmcts-m = [
     "packaging>=23",
 ]
 
+vis-basic = [
+    "Pillow>=10.0.0",
+    "graphviz>=0.20.3",
+    "pyyaml>=6.0",
+]
+
+vis-interactive = [
+    "jinja2>=3.1.0",
+]
+
+vis-all = [
+    "Pillow>=10.0.0",
+    "graphviz>=0.20.3",
+    "jinja2>=3.1.0",
+    "pyyaml>=6.0",
+]
+
```

Now that we have four extras in total, it might be a good time to add [all] extra.

(Actually, there is a [self-referential optional dependency feature](https://hynek.me/articles/python-recursive-optional-dependencies/) for pip>=21.2, but in case the user's pip is super old, I prefer addding all the packages to [all] extra by hand)

```suggestion
all = [
    "jax>=0.5.2",
    "numpy>=1.26.4",
    "numpyro>=0.17.0",
    "pandas>=2.2.3",
    "pymc>=5.21.0",
    "xarray>=2025.1.2",
    "packaging>=23",
    "Pillow>=10.0.0",
    "graphviz>=0.20.3",
    "jinja2>=3.1.0",
    "pyyaml>=6.0",
]

```